packages/gatsby-cli/src/structured-errors/error-map.ts

```diff
@@ -206,24 +214,28 @@ const errors = {
     },
     type: Type.GRAPHQL,
     level: Level.ERROR,
+    category: ErrorCategory.USER,
   },
   "85921": {
     text: (context): string =>
       `There was an error in your GraphQL query:\n\n${context.sourceMessage}\n\nIf you're e.g. filtering for specific nodes make sure that you choose the correct field (that has the same type "${context.inputType}") or adjust the context variable to the type "${context.expectedType}".`,
     type: Type.GRAPHQL,
     level: Level.ERROR,
+    category: ErrorCategory.USER,
   },
```

Comment:
This error can be caused by inference. When all types of a specific field change this might occur. So, in theory, it's user error but it's also part of bad source plugins that do not use schema customization.